extension/src/popup/components/manageAssets/ManageAssetRows/index.tsx

```diff
@@ -182,20 +199,70 @@ export const ManageAssetRows = ({
     return found.length > 0;
   };
 
-  const handleBlockedDomain = (assetData: AssetRowData) => {
-    dispatch(showBlockedDomainWarning());
-    setScamAssetData(assetData);
+  const checkForSuspiciousAsset = async (
+    code: string,
+    issuer: string,
+    domain: string,
+  ): Promise<NewAssetFlags> => {
+    // check revocable
+    let isRevocable = false;
+    try {
+      const resp = await server.assets().forCode(code).forIssuer(issuer).call();
+      isRevocable = resp.records[0]
+        ? resp.records[0].flags.auth_revocable
+        : false;
```

Comment:
there's actually a small edge case. If for some reason the shape of `resp.records[0]` changed and there wasn't a `flags` object anymore, it would cause an error

It would be a little safer to do:

```
isRevocable  = resp?.records[0]?.flags?.auth_revocable
        || false
```

Reply:
nice idea 👍 https://github.com/stellar/freighter/pull/634/commits/a56954289da8e76c17a1d984d6d54ece5909c2a7